public/js/actions/pause.js

```diff
@@ -18,8 +22,22 @@ function resumed() {
  * Debugger has just paused
  */
 function paused(pauseInfo) {
-  return ({ dispatch }) => {
-    return dispatch({
+  return ({ dispatch, threadClient }) => {
+    // If paused by an explicit interrupt, which are generated by the
+    // slow script dialog and internal events such as setting
+    // breakpoints, ignore the event.
+    if(pauseInfo.why.type === "interrupted" &&
+       !pauseInfo.why.onNext) {
+      return;
+    }
+
+    dispatch({
+      type: constants.LOAD_FRAMES,
+      [PROMISE]: promisify(threadClient, threadClient.getFrames,
+                           0, CALL_STACK_PAGE_SIZE)
+    });
+
```

Comment:
Maybe also move the selectSource action here too https://github.com/jlongster/debugger.html/blob/master/public/js/client.js#L49